lib/buffer.js

```diff
@@ -158,18 +157,16 @@ function assertSize(size) {
  **/
 Buffer.alloc = function(size, fill, encoding) {
   assertSize(size);
-  if (size <= 0)
-    return createBuffer(size);
-  if (fill !== undefined) {
+  if (size > 0 && fill !== undefined) {
```

Comment:
@ChALkeR I know there’s the extra `fill()` in there. But if you say it’s faster, I believe that :)